pkg/storage/storage.go

```diff
@@ -70,24 +71,45 @@ type Store interface {
 	GetVolumes(ctx context.Context, accountAddress, asset string) (core.Volumes, error)
 	CountAccounts(context.Context, AccountsQuery) (uint64, error)
 	GetAccounts(context.Context, AccountsQuery) (sharedapi.Cursor[core.Account], error)
-
 	GetBalances(context.Context, BalancesQuery) (sharedapi.Cursor[core.AccountsBalances], error)
 	GetBalancesAggregated(context.Context, BalancesQuery) (core.AssetsBalances, error)
-
-	AppendLog(ctx context.Context, log ...core.Log) error
 	LastLog(ctx context.Context) (*core.Log, error)
 	Logs(ctx context.Context) ([]core.Log, error)
-
 	LoadMapping(ctx context.Context) (*core.Mapping, error)
+
+	UpdateTransactionMetadata(ctx context.Context, id uint64, metadata core.Metadata, at time.Time) error
+	UpdateAccountMetadata(ctx context.Context, id string, metadata core.Metadata, at time.Time) error
+	Commit(ctx context.Context, txs ...core.Transaction) error
 	SaveMapping(ctx context.Context, m core.Mapping) error
-	Initialize(context.Context) (bool, error)
 	Name() string
+}
+
+type Store interface {
+	API
+	Initialize(context.Context) (bool, error)
 	Close(context.Context) error
+	WithTX(ctx context.Context, callback func(api API) error) error
```

Comment:
I can't figure out what this method is doing 😅

Reply:
Create a new transaction and run a callback.
The API interface embed automatically the sql transaction.